### PR TITLE
[Zlib] Falls back to use BouncyCastle if BCL doesn't support

### PR DESCRIPTION
**Pull request by scott-xu**

### CODE REVIEW

File: appveyor.yml [hunk 1]
```diff
@@ -21,12 +21,14 @@ for:
     - echo build
     - dotnet build -f net8.0 -c Debug test/Renci.SshNet.Tests/Renci.SshNet.Tests.csproj
     - dotnet build -f net8.0 -c Debug test/Renci.SshNet.IntegrationTests/Renci.SshNet.IntegrationTests.csproj
+    - dotnet build -f net48 -c Debug test/Renci.SshNet.IntegrationTests/Renci.SshNet.IntegrationTests.csproj
```
Comment (WojciechNagorski) [inline appveyor.yml:24]: How do you run .NET Framework on Linux? :) You should run it on Windows.

File: appveyor.yml [hunk 2]
```diff
@@ -21,12 +21,14 @@ for:
...
     - sh: echo "Run integration tests"
     - sh: dotnet test -f net8.0 -c Debug --no-restore --no-build --results-directory artifacts --logger Appveyor --logger "console;verbosity=normal" --logger "liquid.md;LogFileName=linux_integration_test_net_8_report.md" -p:CollectCoverage=true -p:CoverletOutputFormat=cobertura -p:CoverletOutput=../../artifacts/linux_integration_test_net_8_coverage.xml test/Renci.SshNet.IntegrationTests/Renci.SshNet.IntegrationTests.csproj
+    - sh: dotnet test -f net48 -c Debug --no-restore --no-build --results-directory artifacts --logger Appveyor --logger "console;verbosity=normal" --logger "liquid.md;LogFileName=linux_integration_test_net_48_report.md" -p:CollectCoverage=true -p:CoverletOutputFormat=cobertura -p:CoverletOutput=../../artifacts/linux_integration_test_net_48_coverage.xml --filter Name~Zlib test/Renci.SshNet.IntegrationTests/Renci.SshNet.IntegrationTests.csproj
```
Comment (WojciechNagorski) [inline appveyor.yml:31]: .NET Framework 4.8 on Linux.

**Review by WojciechNagorski — CHANGES_REQUESTED** [review]
You can't run .NET Framework 4.8 test on Linux, can you?

**Comment by scott-xu** [comment]
That's a very good question. I don't know why but the CI works.

**Comment by scott-xu** [comment]
It might be based on Mono.

**Comment by scott-xu** [comment]
Here's CI output

```
dotnet test -f net48 -c Debug --no-restore --no-build --results-directory artifacts --logger Appveyor --logger "console;verbosity=normal" --logger "liquid.md;LogFileName=linux_integration_test_net_48_report.md" -p:CollectCoverage=true -p:CoverletOutputFormat=cobertura -p:CoverletOutput=../../artifacts/linux_integration_test_net_48_coverage.xml --filter Name~Zlib test/Renci.SshNet.IntegrationTests/Renci.SshNet.IntegrationTests.csproj
Test run for /home/appveyor/projects/ssh-net/test/Renci.SshNet.IntegrationTests/bin/Debug/net48/Renci.SshNet.IntegrationTests.dll (.NETFramework,Version=v4.8)
Microsoft (R) Test Execution Command Line Tool Version 17.9.0 (x64)
Copyright (c) Microsoft Corporation.  All rights reserved.
Starting test execution, please wait...
Appveyor.TestLogger: Logging to http://localhost:42353/
A total of 1 test files matched the specified pattern.
  Passed ZlibOpenSsh [596 ms]
Appveyor.TestLogger: 1 test results reported (1 enqueued).
Test Run Successful.
Total tests: 1
     Passed: 1
 Total time: 20.7936 Seconds
Saved report to: linux_integration_test_net_48_report.md
Calculating coverage result...
  Generating report '../../artifacts/linux_integration_test_net_48_coverage.net48.xml'
+--------------------------------+--------+--------+--------+
| Module                         | Line   | Branch | Method |
+--------------------------------+--------+--------+--------+
| Renci.SshNet                   | 17.89% | 12.91% | 24.19% |
+--------------------------------+--------+--------+--------+
| Renci.SshNet.TestTools.OpenSSH | 0%     | 0%     | 0%     |
+--------------------------------+--------+--------+--------+
+---------+--------+--------+--------+
|         | Line   | Branch | Method |
+---------+--------+--------+--------+
| Total   | 17.47% | 12.16% | 23.54% |
+---------+--------+--------+--------+
| Average | 8.94%  | 6.45%  | 12.09% |
+---------+--------+--------+--------+
```

**Comment by WojciechNagorski** [comment]
You're right. There is mono https://www.appveyor.com/docs/linux-images-software/

**Review by WojciechNagorski — APPROVED** [review]
LGTM

**Pull request merged**